File: Python For Everyone, Horstmann/Chapter4-Loops/P4.10.py
# Write a program that reads a word and prints the number of vowels in the word. For
# this exercise, assume that  a e i o u y are vowels. For example, if the user pro vides the
# input  "Harry" , the program prints  2 vowels .

#  situations when Y is not a vowel are not handled.
inputWord = str(input("Enter a word: "))
vowels = ('a', 'e', 'i', 'o', 'u', 'y', 'A', 'E', 'I', 'O', 'U', 'Y')

numVowels = 0

for i in range(len(inputWord)):
    if inputWord[i] in vowels:
        numVowels += 1

print("%d vowels" % numVowels)

File: Java-for-Everyone-Late-Objects-Horstmann/Chapter-4-Loops/P4_10.java
/*Write a program that reads a word and prints the number of vowels in the word. For
this exercise, assume that  a e i o u y are vowels. For example, if the user provides
the input  "Harry" , the program prints  2 vowels.*/

import java.util.Scanner;

public class P4_10 {
    public static void main(String[] args) {
	Scanner input = new Scanner(System.in);
	System.out.print("Word: ");
	String word = input.next();
	input.close();
	int n_vowels = 0;
	for (int i = 0; i < word.length(); i++) {
	    char current_letter = Character.toLowerCase(word.charAt(i));
	    if (current_letter == 'a' || current_letter == 'e' || current_letter == 'o'
                || current_letter == 'i' || current_letter == 'u' || current_letter == 'y') {
		n_vowels += 1;
	    }
	}
	System.out.println("Number of vowels: " + n_vowels);
    }
}
